choral/libs/choral-reactive/src/main/java/choral/reactive/Session.java

```diff
@@ -3,10 +3,13 @@
 import java.util.Random;
 import java.io.Serializable;
 
+/**
+ * A unique identifier for an instance of a choreography at runtime.
+ */
 public class Session implements Serializable {
 
     protected final String choreographyID;
-    protected final String sender;
+    protected final String sender; // TODO The sender should not be here, but some of the logic in ReactiveServer depends on it
```

Comment:
The sender is needed as a part of each message, since all clients connect to the same server, there are no other way to get this information.

But with that said, I agree that it might be nice to move this information out of the session and into the `Message` class instead.

Reply:
Yeah, I think the sender should just be a separate field in the message.

Reply:
No need to fix right now.